src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -516,7 +516,15 @@ export class ConnectionDialogService implements IConnectionDialogService {
 			}));
 
 		}
-		this._logService.error(message);
-		this._errorMessageService.showDialog(severity, headerTitle, message, messageDetails, actions);
+
+		let errorMessage = message;
+		if (message.includes('could not translate host name')) {
+			errorMessage = message.concat('\nCauses:\nUsing the wrong hostname or problems with DNS resolution.\nSuggestions:\nCheck that the server address or hostname is the full address.');
```

Comment:
localize these

But also, are these errors specific to PG? While the problems are generic enough we should avoid having provider-specific logic in the core code itself. Did you look into adding these details into the connection failure code in PGToolsService?